jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -430,89 +435,133 @@ private void checkImageInTargetRegistry() {
+                  results.builtImagesAndBaseImages.get().keySet().iterator().next();
+              return new CheckImageStep(
+                      buildContext,
+                      childProgressDispatcherFactory,
+                      results.targetRegistryClient.get(),
+                      Verify.verifyNotNull(
+                              results
+                                  .builtImagesAndContainerConfigurationPushResults
+                                  .get()
+                                  .get(builtImage))
+                          .get(),
+                      builtImage.get())
+                  .call();
+            });
   }
 
   private void pushImages() {
     ProgressEventDispatcher.Factory childProgressDispatcherFactory =
         Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
 
-    results.buildResult =
+    results.buildResults =
         executorService.submit(
             () -> {
-              realizeFutures(results.baseImageLayerPushResults.get().get(0));
               realizeFutures(results.applicationLayerPushResults.get());
 
-              List<Future<BuildResult>> manifestPushResults =
-                  scheduleCallables(
-                      PushImageStep.makeList(
-                          buildContext,
-                          childProgressDispatcherFactory,
-                          results.targetRegistryClient.get(),
-                          Verify.verifyNotNull(
-                                  results
-                                      .builtImagesAndContainerConfigurationPushResults
+              List<Future<BuildResult>> buildResults = new ArrayList<>();
+
+              for (Map.Entry<Future<Image>, Image> entry :
+                  results.builtImagesAndBaseImages.get().entrySet()) {
+
+                Future<Image> builtImage = entry.getKey();
+                Image baseImage = entry.getValue();
+
+                Future<BuildResult> buildResult =
+                    executorService.submit(
+                        () -> {
+                          realizeFutures(
+                              Verify.verifyNotNull(
+                                  results.baseImagesAndLayerPushResults.get().get(baseImage)));
+
+                          List<Future<BuildResult>> manifestPushResults =
+                              scheduleCallables(
+                                  PushImageStep.makeList(
+                                      buildContext,
+                                      childProgressDispatcherFactory,
+                                      results.targetRegistryClient.get(),
+                                      Verify.verifyNotNull(
+                                              results
+                                                  .builtImagesAndContainerConfigurationPushResults
+                                                  .get()
+                                                  .get(builtImage))
```

Comment:
This is used on line 500, so I suggest refactor out into a local variable. I think it will considerable improve readability.